filebeat/module/iis/access/_meta/fields.yml

```diff
@@ -82,87 +37,91 @@
       format: bytes
       description: >
         The number of bytes of the server request body.
-    - name: agent
-      type: text
-      description: >
-        Contains the un-parsed user agent string. Only present if the user
-        agent Elasticsearch plugin is not available or not used.
+
+    - name: server_ip
+      type: alias
+      path: destination.ip
+    - name: method
+      type: alias
+      path: http.request.method
+    - name: url
+      type: alias
+      path: url.path
+    - name: query_string
+      type: alias
+      path: url.query
+    - name: port
+      type: alias
+      path: destination.port
+    - name: user_name
+      type: alias
+      path: user.name
+    - name: remote_ip
+      type: alias
+      path: source.ip
+    - name: referrer
+      type: alias
+      path: http.request.referrer
+    - name: response_code
+      type: alias
+      path: http.response.status_code
+    - name: http_version
+      type: alias
+      path: http.version
+    - name: hostname
+      type: alias
+      path: host.hostname
     - name: user_agent
       type: group
-      description: >
-        Contains the parsed user agent field. Only present if the user
-        agent Elasticsearch plugin is available and used.
       fields:
         - name: device
-          type: keyword
-          description: >
-            The name of the physical device.
+          type: alias
+          path: user_agent.device
         - name: major
-          type: long
-          description: >
-            The major version of the user agent.
+          type: alias
+          path: user_agent.major
         - name: minor
-          type: long
-          description: >
-            The minor version of the user agent.
+          type: alias
+          path: user_agent.minor
         - name: patch
-          type: keyword
-          description: >
-            The patch version of the user agent.
+          type: alias
+          path: user_agent.patch
         - name: name
-          type: keyword
-          example: Chrome
-          description: >
-            The name of the user agent.
+          type: alias
+          path: user_agent.name
         - name: os
-          type: keyword
-          description: >
-            The name of the operating system.
+          type: alias
+          path: user_agent.os.full_name
         - name: os_major
-          type: long
-          description: >
-            The major version of the operating system.
+          type: alias
+          path: user_agent.os.major
```

Comment:
We could probably use script processor to concatenate the version fields. Initially hoped the the join processor would do this but seems it has a different purpose.

Reply:
Yeah, but all of this refining needs to go in a shared pipeline. We're not going to duplicate this work everywhere, on the eve of starting to share pipelines ;-)